coffee/web_socket_channel.coffee

```diff
@@ -36,12 +36,14 @@ class palava.WebSocketChannel extends @EventEmitter
     @socket.onmessage = (msg) =>
       try
         parsedMsg = JSON.parse(msg.data)
-        if parsedMsg.event == "pong"
-          @outstandingPongs = 0
-        else
-          @emit 'message', parsedMsg
       catch SyntaxError
-        @emit 'error', 'invalid_json', msg
+        @emit 'error', 'invalid_json', msg.data
+        return
+
+      if parsedMsg.event == "pong"
```

Comment:
parsedMsg?.event maybe?

Reply:
if the guard did not catch it, we are good and parsedMsg is defined

Reply:
but is it clear that this is an object? what if the json string is "0"? (I know with a server that adheres to the protocol that should not happen) - so because it is unlikely we could also leave it this way of course